.azure-pipelines/benchmarks.yml

```diff
@@ -7,20 +7,8 @@ trigger:
 variables:
   buildConfiguration: release
   dotnetCoreSdkVersion: 5.0.100
-  ddApiKey: $(DD_API_KEY)
```

Comment:
Is it possible to just make it conditional on `DD_API_KEY` not being empty? My attempt to conditionally disable the `Arm64` pool failed but want to be sure that it also doesn't work here.

Reply:
What would be the benefit such approach? I see value of doing it if we would like to enable it (provide the API key variable value). Otherwise it feels for me like unnecessary waste for anyone reading the code.

Of course I can do that on Monday if there are any other reasons or you insist.

Moreover, take into consideration that merging it even now will at least make unit tests green in Azure DevOps and would give more confidence in other PRs ;)

Reply:
By using a conditional the files can be identical between both repos. In general, keeping the diff smaller between the repos is a benefit, a modest one, in this specific case, but worth keeping in mind.

I don't think it is a must but desirable. I wouldn't block this change on this specific issue.

Reply:
I can give it a try with conditional and make a second PR on Monday 🙂

Reply:
I have not found any possibility to conditionally use a container/service https://docs.microsoft.com/en-us/azure/devops/pipelines/process/resources?view=azure-devops&tabs=schema#resources-containers 

I tried to disable the `HEALTHCHECK` of the dd_agent but it is also not possible:
- https://docs.microsoft.com/en-us/azure/devops/pipelines/process/service-containers?view=azure-devops&tabs=yaml#healthcheck
- https://developercommunity.visualstudio.com/content/problem/1164638/cant-change-service-container-healthcheck-timings.html

Here is a PR that disables Datadog Agent with minimal changes: TBD

Reply:
Thanks for trying @pellared. Unfortunately, it seems that the options in Az Pipelines don't allow an easy disable for this. Let's go with your current solution.